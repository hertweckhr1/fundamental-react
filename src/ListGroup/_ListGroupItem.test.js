/* eslint-disable jsx-a11y/anchor-is-valid */
import { Button } from '../';
import ListGroup from './ListGroup';
import ListGroupItem from './_ListGroupItem';
import { mount } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';

describe('<ListGroupItem />', () => {
    const defaultListGroupItem = (
        <ListGroupItem className='blue'>
            <a style={{ cursor: 'pointer' }}>List item 1</a>
        </ListGroupItem>
    );

    test('create list group', () => {
        // create default list group
        let component = renderer.create(defaultListGroupItem);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    describe('Prop spreading', () => {

        test('should allow props to be spread to the ListGroupItem component', () => {
            const element = mount(<ListGroupItem data-sample='Sample' />);

            expect(
                element.getDOMNode().attributes['data-sample'].value
            ).toBe('Sample');
        });

    });
});
