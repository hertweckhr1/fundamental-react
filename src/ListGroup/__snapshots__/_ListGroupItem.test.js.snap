// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListGroupItemActions /> create list group 1`] = `
<li
  className="fd-list-group__item blue"
>
  <a
    style={
      Object {
        "cursor": "pointer",
      }
    }
  >
    List item 1
  </a>
</li>
`;
